pertpy/tools/_metadata.py

```diff
@@ -0,0 +1,118 @@
+            _download(
+                url="https://ndownloader.figshare.com/files/35020903",
+                output_file_name=cell_line_file_name,
+                output_path=settings.cachedir,
+                is_zip=False,
+            )
+            self.cell_line_meta = pd.read_csv(settings.cachedir.__str__() + "/" + cell_line_file_name)
+        else:
+            self.cell_line_meta = pd.read_csv(cell_line_file_path)
+
+    def annotate_cell_lines(
+        self,
+        adata: anndata,
+        cell_line_column: str = "DepMap_ID",
+        cell_line_type: str = "DepMap_ID",
+        cell_line_information: list[str] = None,
+        copy: bool = False,
+    ) -> anndata:
+        """Fetch cell line annotation.
+
+        For each cell, we fetch cell line annotation from Dependency Map (DepMap).
+
+        Args:
+            adata: The data object to annotate.
+            cell_line_column: The column of `.obs` with cell line information. (default: "DepMap_ID")
+            cell_line_type: The type of cell line information, e.g. DepMap_ID, cell_line_name or stripped_cell_line_name. (default: "DepMap_ID")
+            cell_line_information: The metadata to fetch. All metadata will be fetched by default. (default: all)
+            copy: Determines whether a copy of the `adata` is returned. (default: False)
+
+        Returns:
+            Returns an AnnData object with cell line annotation.
+        """
+        if copy:
+            adata = adata.copy()
+        if cell_line_type in self.cell_line_meta.columns:
+            """If the specified cell line type can be found in the DepMap database,
+            we can compare these keys and fetch the corresponding metadata."""
+            if cell_line_information is None:
+                """If no cell_line_information is specified, all metadata is fetched by default.
+                Sometimes there is already different cell line information in the `adata`.
+                In order to avoid redundant information,
+                we will remove the duplicate information from metadata after merging."""
+                adata.obs = (
+                    adata.obs.reset_index()
+                    .merge(
+                        self.cell_line_meta,
+                        left_on=cell_line_column,
+                        right_on=cell_line_type,
+                        how="left",
+                        suffixes=("", "_fromMeta"),
+                    )
+                    .filter(regex="^(?!.*_fromMeta)")
+                    .set_index(adata.obs.index.names)
+                )
+                """ If cell_line_column and cell_line_type have different names,
+                there will be a column for each of them after merging,
+                which is redundant as they refer to the same information.
+                We will move the cell_line_type column."""
+                if cell_line_column != cell_line_type:
+                    del adata.obs[cell_line_type]
+
```

Comment:
Somewhere here there should be support for adding information such as bulk transcriptomics, methylation, protein exp., etc. into `.obsm`. For instance, the `CCLE_expression.csv` file on DepMap contains bulk RNA expression values which would be useful to load in as well, all under the category of DepMap data.